HoloPipelines/pipelineServer.py

```diff
@@ -0,0 +1,202 @@
+from flask_json import json_response
+from datetime import datetime
+from pathlib import Path
+import uuid
+import json
+import threading
+import time
+import logging
+import requests
+import os
+import pathlib
+from zipfile import ZipFile
+
+app = Flask(__name__)
+app.config["JSON_ADD_STATUS"] = False
+
+# global variables
+this_cwd = pathlib.Path.cwd()
+save_directory = Path("input")
+output_directory = Path("output")
+pipeline_list = getPipelineList()
+status = {
+    "j0": {"status": "segment", "stamp": "2019-08-05 14:09:19"},
+    "j1": {"status": "segment", "stamp": datetime.now().strftime("%Y-%m-%d %H:%M:%S")},
+}
+jobid2plid = {"j0": "nifti2glb", "j1": "dicom2glb"}
+job2inputURL = {"j0": "url"}
+job2outputURL = {}
+
+
+# logging formatting
+FORMAT = "%(asctime)-15s -function name:%(funcName)s -%(message)s"
+logging.basicConfig(level=logging.INFO, format=FORMAT)
+
+
+def save_request_file_to_input_dir(filename, request_input_data_URL):
+    global save_directory
+    if not save_directory.is_dir():
+        os.mkdir("input")
+    response = requests.get(request_input_data_URL)
+    if response.status_code != 200:
+        return json_response(jobID="image study endpoint failure", status_code=404)
+    file_dir = this_cwd.joinpath(str(save_directory), filename)
+    open(file_dir, "wb+").write(response.content)
+    logging.info("file dir: " + str(file_dir))
+
+
+def unzip_the_file(filename):
+    # get unzip input filename
+    filename_unzip = filename.rsplit(".", 1)[0]
+    logging.info("unzip file name:" + filename_unzip)
+
+    # unzip the input file
+    file_dir = this_cwd.joinpath(str(save_directory), filename)
+    path_to_store_unzip_file = this_cwd.joinpath(str(save_directory), filename_unzip)
+    with ZipFile(str(file_dir), "r") as zipObj:
+        # unzip
+        zipObj.extractall(str(path_to_store_unzip_file))
+    return str(path_to_store_unzip_file)
+
+
+# cleaning the status
+@app.before_first_request
+def activate_status_cleaning_job():
+    def run_job():
+        while True:
+            global status
+            # logging.info("status before loop: " +str(status))
+
+            for job in status.copy():
+
+                job_time_string = status[job]["stamp"]
+                # logging.info("job-time: "+job_time_string)
+
+                job_time_obj = datetime.strptime(job_time_string, "%Y-%m-%d %H:%M:%S")
+
+                current_time = datetime.now()
+                delta_time = (current_time - job_time_obj).total_seconds()
+                # logging.info("time difference: "+str(delta_time))
+
+                # if job exists more than 30 mins delete it from dictionary
+                if delta_time >= 1800.0:
+                    status.pop(job)
+
+            time.sleep(30)
+
+    # logging.info("status after loop: "+str(status))
+    thread = threading.Thread(target=run_job)
+    thread.start()
+
+
+# update the status from pipeline
+@app.route("/api/v1/status", methods=["POST"])
```

Comment:
There are actually more elegang ways to have the same prefix for all routes, which also makes it easier if it changes. See https://stackoverflow.com/questions/18967441/add-a-prefix-to-all-flask-routes